nni/retiarii/hub/pytorch/proxylessnas.py

```diff
@@ -133,7 +132,8 @@ def __init__(
             ConvBNReLU(in_channels, out_channels, kernel_size=1, norm_layer=norm_layer, activation_layer=nn.Identity)
         ]
         super().__init__(*simplify_sequential(blocks))
-        self.has_skip = stride == 1 and in_channels == out_channels
+        # NOTE: "is" is used here instead of "==" to avoid creating a new value choice.
```

Comment:
`ValueChoice([1,2,3]) == ValueChoice([2,3,4])` will create a new ValueChoice, which is true iff. two value choices choose the same value.

We could use techniques like "expression simplify" to simplify those expressions which must be true in all conditions. But I think that's a bit too complex.

Reply:
if two value choices `a` `b` have the same label, the `a is b` statement is True or False?

BTW, it would be better to add comments in a proper place to explain the behavior of `is` and `==` of value choice

Reply:
> if two value choices a b have the same label, the a is b statement is True or False?

No. `is` is reference equality. It only returns true when the underlying pointer points to the same memory address.

I think people will notice this when debugging. When they accidentally use `==` by mistake, it will raise an error here because it's found to be a value choice. `is`, on the other hand, is not what you will commonly do in daily life. It's just a hack here to support two types of channels: both int and value choice of int.